packages/platform-android/src/commands/runAndroid/adb.ts

```diff
@@ -41,6 +41,57 @@ function getDevices(adbPath: string): Array<string> {
   }
 }
 
+function getAllAvailableDevices(adbPath: string) {
+  const emulators = execFileSync(
+    `${process.env.ANDROID_HOME}/emulator/emulator`,
+    ['-list-avds'],
+    {encoding: 'utf8'},
+  )
+    .replace(/\n$/, '')
+    .split('\n')
+    .filter(Boolean)
+    .map(emulator => ({
+      deviceId: undefined,
+      name: emulator,
+      type: 'emulator',
+    }));
+
+  const physicalDevicesList = execFileSync(adbPath, ['devices', '-l'], {
+    encoding: 'utf-8',
+  })
+    .toString()
+    .replace(/\n$/, '')
+    .split('\n');
+
+  const modelKey = 'model:';
+  const physicalDevices = physicalDevicesList
+    .slice(1, physicalDevicesList.length)
+    .map((device: string) => {
+      if (device.includes('emulator')) {
+        return;
+      }
+
+      const [deviceId, ...deviceInfo] = device.split(' ');
+
+      if (!deviceId) {
+        return;
+      }
+
```

Comment:
nit: It seems to me there's a bit too much of line breaks here, following your other code in other places